aten/src/ATen/native/BatchLinearAlgebraKernel.cpp

```diff
@@ -55,13 +65,23 @@ std::tuple<Tensor, Tensor> eig_kernel_impl(const Tensor& self, bool& eigenvector
   self_.copy_(self);
 
   auto options = self.options().memory_format(LEGACY_CONTIGUOUS_MEMORY_FORMAT);
-  Tensor vals_ = at::empty_strided({n, 2}, {1, n}, options);
+
+  // the API is slightly different for the complex vs real case: if the input
+  // is complex, eigenvals will be a vector of complex. If the input is real,
+  // eigenvals will be a (n, 2) matrix containing the real and imaginary parts
```

Comment:
As discussed in #43081, I think we should always return a complex tensor. This will be a bc breaking change, so we should only update this behavior for `torch.linalg.eig`

Reply:
I thought that the point of `torch.linalg.*` was to be as compatible to numpy as possible. This would be an unnecessary breakage for people porting their code from numpy to pytorch.
What about adding a flag such as `returns_complex=False` (or `=True` if we want the correct-but-numpy-incompatible behavior by default) to let the user choose?

Reply:
Hey, @antocuni , looks like Numpy is already doing it:
```
In [18]: x = numpy.array([[1, 2], [-2, 1]], numpy.double)

In [19]: x
Out[19]: 
array([[ 1.,  2.],
       [-2.,  1.]])

In [20]: numpy.linalg.eig(x)
Out[20]: 
(array([1.+2.j, 1.-2.j]),
 array([[0.        -0.70710678j, 0.        +0.70710678j],
        [0.70710678+0.j        , 0.70710678-0.j        ]]))

```

Reply:
ouch, my fault, I overlooked it. Ok, I'll implement the "proper" behavior than, thanks for pointing it out

Reply:
@anjali411 @nikitaved I just realized that `torch.linalg.eig` doesn't exists yet!
So I think that the current behavior of my branch is correct. I agree that `torch.linalg.eig` should always return complex, but I don't think there is anything we can do in this branch